src/gui/shareusergroupwidget.cpp

```diff
@@ -416,6 +419,71 @@ ShareUserLine::ShareUserLine(QSharedPointer<Share> share,
     if (!share->account()->capabilities().shareResharing()) {
         _ui->permissionShare->hide();
     }
+
+    loadAvatar();
+}
+
+void ShareUserLine::loadAvatar()
+{
+    const int avatarSize = 48;
+
+    // Set size of the placeholder
+    _ui->avatar->setMinimumHeight(avatarSize);
+    _ui->avatar->setMinimumWidth(avatarSize);
+    _ui->avatar->setMaximumHeight(avatarSize);
+    _ui->avatar->setMaximumWidth(avatarSize);
+    _ui->avatar->setAlignment(Qt::AlignCenter);
+
+    /* Create the fallback avatar.
+     *
+     * It's created the same way as on the webinterface to make
+     * the colors match.
+     *
+     * This will be shown until the avatar image data arrives.
+     */
+    QString seed = _share->getShareWith()->shareWith();
+    if (_share->getShareWith()->type() != Sharee::User) {
+        seed += QString(" %1").arg(_share->getShareWith()->type());
+    }
+
+    const QByteArray hash = QCryptographicHash::hash(seed.toUtf8(), QCryptographicHash::Md5);
+    int hue = ((double)hash.mid(0, 3).toHex().toInt(0, 16) / (double)0xffffff) * 255;
```

Comment:
nitpick: C cast

I guess this could be rewriten as int(hash[0]),  so we don't do two allocations and some non-trivial computation.